interface/ceed.c

```diff
@@ -751,6 +788,81 @@ int CeedReference(Ceed ceed) {
   return CEED_ERROR_SUCCESS;
 }
 
+/**
+  @brief Get a `CeedVector` for scratch work from a `Ceed` context.
+
+  Note: This vector must be restored with @ref CeedRestoreWorkVector().
+
+  @param[in]  ceed `Ceed` context
+  @param[in]  len  Minimum length of work vector
+  @param[out] vec  Address of the variable where `CeedVector` will be stored
+
+  @return An error code: 0 - success, otherwise - failure
+
+  @ref Backend
+**/
+int CeedGetWorkVector(Ceed ceed, CeedSize len, CeedVector *vec) {
+  CeedInt i = 0;
+
+  if (!ceed->work_vectors) CeedCall(CeedWorkVectorsCreate(ceed));
+
+  // Search for big enough work vector
+  for (i = 0; i < ceed->work_vectors->num_vecs; i++) {
+    if (!ceed->work_vectors->is_in_use[i]) {
+      CeedSize work_len;
+
+      CeedCall(CeedVectorGetLength(ceed->work_vectors->vecs[i], &work_len));
+      if (work_len >= len) break;
+    }
+  }
+  // Long enough vector was not found
+  if (i == ceed->work_vectors->num_vecs) {
+    if (ceed->work_vectors->max_num_vecs == 0) {
+      ceed->work_vectors->max_num_vecs = 1;
+      CeedCall(CeedCalloc(ceed->work_vectors->max_num_vecs, &ceed->work_vectors->vecs));
+      CeedCall(CeedCalloc(ceed->work_vectors->max_num_vecs, &ceed->work_vectors->is_in_use));
+    } else if (ceed->work_vectors->max_num_vecs == i) {
+      ceed->work_vectors->max_num_vecs *= 2;
```

Comment:
`max_num_vecs` makes me think the variable should be static (i.e. the number of work vectors before we start to destroy the last untouched one). I can't think of a better name though.

Reply:
Naming aside, having the number of stored vectors always increase seems like it might be a foot-gun at some later point. I feel like we should have a hard limit before it starts to cull the stored vecs.

Granted, I don't see exponential growth of stored vecs being an issue, but just a thought.

Reply:
We can pick a limit, but I guess it feels arbitrary to me since each vector can be arbitrarily large already.